types/xast/index.d.ts

```diff
@@ -0,0 +1,107 @@
+// Type definitions for non-npm package xast 1.0
+// Project: https://github.com/syntax-tree/xast
+// Definitions by: stefanprobst <https://github.com/stefanprobst>
+//                 Titus Wormer <https://github.com/wooorm>
+//                 Christian Murphy <https://github.com/ChristianMurphy>
+//                 Junyoung Choi <https://github.com/rokt33r>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+import { Parent as UnistParent, Literal as UnistLiteral, Node as UnistNode } from 'unist';
+
+export { UnistNode as Node };
+
+/**
+ * Node in xast containing other nodes.
+ * Its content is limited to only other xast content.
+ */
+export interface Parent extends UnistParent {
+    children: Array<Element | Text | Comment | DocType | Instruction | Cdata>;
+}
+
+/**
+ * Node in xast containing a value.
+ */
+export interface Literal extends UnistLiteral {
+    value: string;
+}
+
+/**
+ * Document fragment or a whole document.
+ * Should be used as the root of a tree and must not be used as a child.
+ *
+ * XML specifies that documents should have exactly one element child,
+ * therefore a root should have exactly one element child when representing a
+ * whole document.
+ */
+export interface Root extends Parent {
+    type: 'root';
+}
+
+/**
+ * An XML element.
+ */
+export interface Element extends Parent {
+    type: 'element';
+    /**
+     * The element's qualified name.
+     */
+    name: string;
+    /**
+     * Information associated with the element.
+     */
+    attributes?: Attributes;
+    children: Array<Element | Text | Comment | Instruction | Cdata>;
+}
+
+/**
+ * Information associated with an element.
+ */
+export interface Attributes {
+    [name: string]: string;
```

Comment:
are attribute values guaranteed to be string?
`hast` has a similar concept in properties, that supports a wide range of primitives.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c06fa19fdc843beedca332198de946928d24e18e/types/hast/index.d.ts#L72-L77

/cc @wooorm

Reply:
the `hast` docs have `typedef any PropertyValue`, while `xast` docs have `typedef string AttributeValue`, but yes would be good to confirm this

Reply:
Attribute values are indeed strings, unlike hast!

Reply:
Although, you might add `null` / `undefined` too, as those values *should* be handled by utilities